angularSTEP/src/app/star-rating/star-rating.component.ts

```diff
@@ -0,0 +1,52 @@
+import { Component, OnInit, Input, Output, EventEmitter, SimpleChanges } from '@angular/core';
+import {MatSnackBar} from '@angular/material/snack-bar';
+
+@Component({
+  selector: 'app-star-rating',
+  templateUrl: './star-rating.component.html',
+  styleUrls: ['./star-rating.component.scss']
+})
+export class StarRatingComponent implements OnInit {
+  maxRating: number = 5;
+  ratingsArray: boolean[] = Array(this.maxRating).fill(false);
+  @Input() currentRating: number;
+  @Output() newRating = new EventEmitter<number>();
+
+  constructor(private _snackBar: MatSnackBar) { }
```

Comment:
Just curious why this name starts with an underscore

Reply:
I just saw that the Angular snackBar documentation had it like that. I don't think it actually matters and think it is a style thing, but I just decided to go with it.